src/core/server/elasticsearch/client/mocks.ts

```diff
@@ -6,20 +6,23 @@
  * Side Public License, v 1.
  */
 
-import { Client, ApiResponse } from '@elastic/elasticsearch';
+import type { Client, ApiResponse } from '@elastic/elasticsearch';
 import { TransportRequestPromise } from '@elastic/elasticsearch/lib/Transport';
 import type { DeeplyMockedKeys } from '@kbn/utility-types/jest';
 import { ElasticsearchClient } from './types';
 import { ICustomClusterClient } from './cluster_client';
 import { PRODUCT_RESPONSE_HEADER } from '../supported_server_response_check';
 
+// use jest.requireActual() to prevent weird errors when people mock @elastic/elasticsearch
+const { Client: UnmockedClient } = jest.requireActual('@elastic/elasticsearch');
```

Comment:
This fix is done by @spalger. It seems to be a bug in `mockify` breaking runtime code, when it tries to mock an already mocked object. Related issue https://github.com/elastic/kibana/issues/108388